src/cpp/shared/game/game.cpp

```diff
@@ -165,4 +182,77 @@ void philAI::RedistributeTroops(armyGroup *army1, armyGroup *army2, int a1, int
+  randomizedValue2 = Random(0, 99);
+  hasTier[TIER_THREE] = (isAI < 1 ? 25 : 65) > randomizedValue2;// If isAI, there is a ~65% chance that this hero will get Tier 3 creatures; if not, then a ~25% chance
+  if ((isAI < 1 ? 25 : 65) <= randomizedValue2)// If isAI, there is a ~35% chance that this hero will get Tier 2 creatures; if not, then a ~75% chance
+    hasTier[TIER_TWO] = 1;
+  for (armySlotIdx = 0; armySlotIdx < 5; ++armySlotIdx) {
+    pointerToArmyOfHeroIdx->creatureTypes[armySlotIdx] = -1;
+    pointerToArmyOfHeroIdx->quantities[armySlotIdx] = -1;
+  }
+  for (creatureTier = 0; creatureTier < 2; ++creatureTier) {
+    if (hasTier[creatureTier]) {
+      pointerToArmyOfHeroIdx->creatureTypes[creatureTier2] = creatureTypesAndQuantities[gpGame->heroes[heroIdx].factionID][creatureTier][TIER_ONE];
+      randomLowerBound = 10 * creatureTypesAndQuantities[gpGame->heroes[heroIdx].factionID][creatureTier][TIER_TWO];
+      randomUpperBound = 10 * creatureTypesAndQuantities[gpGame->heroes[heroIdx].factionID][creatureTier][TIER_THREE] + 9;
+      if (isAI) {                              //  If isAI, randomLowerBound is assigned the average of the bounds and results in the probability of higher random values
+        randomLowerBound = (randomUpperBound + randomLowerBound) / 2;
+      }
+      randomizedValue = Random(randomLowerBound, randomUpperBound) / 10;
+      pointerToArmyOfHeroIdx->quantities[creatureTier2++] = randomizedValue;
+    }
+  }
+}
+
+void game::GiveTroopsToNeutralTown(int castleIdx) {
+  int randomQuantity; // [sp+18h] [bp-14h]@7
+  int quantity; // [sp+18h] [bp-14h]@15
+  signed int creatureTier; // [sp+1Ch] [bp-10h]@7
+  int creatureType; // [sp+24h] [bp-8h]@0
+  int creatureTierRandomizer; // [sp+28h] [bp-4h]@4
+
+  if ((this->castles[castleIdx].x > 0 || this->castles[castleIdx].y > 0) && this->castles[castleIdx].ownerIdx < 0) {
+    creatureTierRandomizer = Random(1, 15);
+    if (giCurTurn / 10) { // As time goes on, the odds of getting smaller stacks of stronger creatures increases, but it is still possible to get larger stacks of weaker creatures
+      creatureTierRandomizer += Random(0, giCurTurn / 10);
+    }
+    if (creatureTierRandomizer <= 5) {
+      creatureTier = 10;
+      randomQuantity = Random(8, 15);
+    } else if (creatureTierRandomizer <= 10) {
+      creatureTier = 20;
+      randomQuantity = Random(5, 7);
+    } else if (creatureTierRandomizer <= 13) {
+      creatureTier = 30;
+      randomQuantity = Random(3, 5);
+    } else if (creatureTierRandomizer <= 15) {
+      creatureTier = 40;
+      randomQuantity = Random(1, 3);
+    } else {
+      creatureTier = 50;
+      randomQuantity = 1;
+    }
+    quantity = giCurTurn / 20 + randomQuantity;
+    creatureType = neutralTownCreatureTypes[this->castles[castleIdx].factionID][creatureTier / 10];
```

Comment:
Hmmm.....so, this function could have been generalized just by changing this array rather than by extracting this function. Oh well.

Reply:
I'm just leaving this here so I can tell my self that this comment has been addressed; I understand that the comment is obsolete now.